src/Evolution/DgSubcell/SubcellOptions.cpp

```diff
@@ -23,7 +35,74 @@ SubcellOptions::SubcellOptions(double initial_data_rdmp_delta0,
       persson_exponent_(persson_exponent),
       always_use_subcells_(always_use_subcells),
       reconstruction_method_(recons_method),
-      use_halo_(use_halo) {}
+      use_halo_(use_halo),
+      only_dg_block_and_group_names_(std::move(only_dg_block_and_group_names)) {
+  if (not only_dg_block_and_group_names_.has_value()) {
+    only_dg_block_ids_ = std::vector<size_t>{};
+  }
+}
+
+template <size_t Dim>
+SubcellOptions::SubcellOptions(
+    const SubcellOptions& subcell_options_with_block_names,
+    const DomainCreator<Dim>& domain_creator)
+    : initial_data_rdmp_delta0_(
+          subcell_options_with_block_names.initial_data_rdmp_delta0_),
+      initial_data_rdmp_epsilon_(
+          subcell_options_with_block_names.initial_data_rdmp_epsilon_),
+      rdmp_delta0_(subcell_options_with_block_names.rdmp_delta0_),
+      rdmp_epsilon_(subcell_options_with_block_names.rdmp_epsilon_),
+      initial_data_persson_exponent_(
+          subcell_options_with_block_names.initial_data_persson_exponent_),
+      persson_exponent_(subcell_options_with_block_names.persson_exponent_),
+      always_use_subcells_(
+          subcell_options_with_block_names.always_use_subcells_),
+      reconstruction_method_(
+          subcell_options_with_block_names.reconstruction_method_),
+      use_halo_(subcell_options_with_block_names.use_halo_),
+      only_dg_block_ids_([&domain_creator](
```

Comment:
I think you could avoid this long list of copies by making this a mutating function like `parse_block_names(domain_creator)`. Just a suggestion.

Reply:
I prefer a constructor

Reply:
Why?

Reply:
It makes the flow clearer, and also allows you to do all of this with a constructor in tests.

Finally, updating all the tests is tedious and sort of a waste of time in this case... I don't see a benefit to not using a constructor other than removing a few lines.

Reply:
Maybe a simple `*this = subcell_options_with_block_names` then to take care of copying everything? Just to make this easier to maintain..